### ISSUE TITLE
Tweak data frame HTML repr

### ISSUE THREAD

**Issue by cbrnr** [pola-rs/polars#18100]
### Description

Data frames are printed as a nice HTML table when working in Jupyter notebook:

```python
import polars as pl

df = pl.DataFrame({
    "a": [1, 2, 3, 4, 5],
    "b": ["A", "B", "C", "D", "E"],
    "c": [1.1, -2.8, 3.4, 4.7, -5.9],
})

df
```

This currently results in the following output:
<img width="124" alt="Screenshot 2024-08-08 at 14 04 37" src="https://github.com/user-attachments/assets/f00e4934-f04c-413c-a767-4089d3d37e02">

I find the styling of the column types a bit distracting though, they look exactly like column headers. Therefore, I suggest to print column types in italic, which would look like this:

<img width="110" alt="Screenshot 2024-08-08 at 14 14 09" src="https://github.com/user-attachments/assets/30d5fada-98a5-4718-be7a-ce89d68ee4cc">

**Comment by cbrnr** [comment]
Will this also get rid of the bold formatting?

**Comment by deanm0000** [comment]
No, I didn't really notice it. I also don't see where the bolding comes in, maybe because it's part of thead.

**Comment by cbrnr** [comment]
It's probably some predefined style of the header row. Would this be a change that everyone would like to see?

**Comment by mcrumiller** [comment]
I would prefer either italics or a 10-pt Consolas (or other fixed-width font) in grey (not black) to slightly subdue it.

**Comment by cbrnr** [comment]
I definitely like using gray to slightly subdue it. I'm not sure about a fixed-width font, I slightly prefer the regular font italicized, but here's what these variants look like:

<img width="637" alt="Screenshot 2024-08-08 at 17 00 35" src="https://github.com/user-attachments/assets/e4eaa4f0-eb60-4fba-849c-266e6e9e98ab">

**Comment by mcrumiller** [comment]
I said 10-pt assuming everything else was 12pt; What about a fixed with a 8- or 9- pt? If that looks no better then yes, from the screenshots above I strongly prefer the italic.

**Comment by mcrumiller** [comment]
Another nit: the `str` columns are center-aligned I believe (I can't tell from your screenshots) but the header is obviously right-aligned. We should be consistent here with header + column alignment.

**Comment by cbrnr** [comment]
I can add more mockups, let me know what you would like to see. Note that I'm not generating them with the original HTML rendering, so they might not look 100% like the real thing.

Re alignment, the entire `str` column is in fact right-aligned, I think it's the added quotes (and the fact that the strings are just single characters) that make it look like it was centered.

**Comment by cbrnr** [comment]
I think it would also be useful to show row numbers, here's a suggestion:

<img width="152" alt="Screenshot 2024-09-10 at 09 37 03" src="https://github.com/user-attachments/assets/e363a75e-75f9-4c03-b296-23d3836381e6">
